src/index.ts

```diff
@@ -267,6 +267,11 @@ export function apply(ctx: Context, config: Config) {
             return { model, input: prompt, parameters: omit(parameters, ['prompt']) }
           }
           case 'sd-webui': {
+            // set default denoising strength to `0.75` for `hires fix` feature
+            // https://github.com/koishijs/novelai-bot/issues/158
+            if (options.hiresFix || config.hiresFix) {
+              parameters.strength ??= 0.75
```

Comment:
It looks wield to just set default strength when hiresFix is enabled. Can this be changed to "when backend is sd" or other behavior?

Reply:
Ohhhh I see. Then why not removing the `if` statement and just set the default value to 0.75?

Reply:
Well, that makes sense too.

Reply:
I think it would be more reasonable to move the if out of the case, and set default value to 0.75 regardless of sd-webui. in case there are more backends to come.

Reply:
But there are only sd-webui provides `hires fix` feature, for now at least.

Reply:
How about stable horde?
<!-- BOT-MESSAGE-FOOTER -->
---
Generated by [koishi-plugin-github](https://github.koishi.chat/).

Reply:
They don't have this feature currently.

Reply:
You see, stable horde is a completely independent application compare to stable diffusion webui, so they have no any even relationship except using stable diffusion algorithms and models.

Reply:
That makes sense. Another concern I have is is this default of 0.75 what most people expect? Will it have a big impact on the resulting image?

Reply:
I DON'T KNOW.

But 0.75 is the default value of sd-webui for generating images with `hires fix` enabled. So I wrote this specific value.

Reply:
It is also possible for users to change the value by adding `--strength` parametres, so I don't think it is a big deal.

As for the `resulting image`, since you are using `hires fix`, it would not gerantee that you could use the same parametres to reproduce the same result with the same seed. If you're referring the content, the second part of `hires fix` would still feeds the same prompts to the diffusion model, so at least it would not show you a dinosaur while you are writing Bulbasaur.

Reply:
> 0.75 is the default value of sd-webui for generating images with hires fix enabled.

> So I think it would be more reasonable to move the if statement out of the case, and set default value to 0.75 regardless of sd-webui.

Other backends which do not support this feature also do not have this option on the first place.

Reply:
Yep. So I think it is good for now.

Reply:
Ehhhh I mean moving the `if` outwards would be ok and furthermore be conducive to decoupling - the place you modified should only be related to request payloads, not default behavior of parameters.

Reply:
But the others backends have their own default `strength` value which are not expectedly the same with `sd-webui`'s. The `strength` we are setting here just an in-case for txt2img api that has default `denoising_strength` set to 0 that causes error. Shouldn't we not to change the default behaviours for other backends?

Reply:
I believe you misunderstood. What do I mean is to **not** delete the `if` statement but move it outwards. In that case, we are certainly not changing the default behaviours for other backends.